sycl/test/basic_tests/fp-accuracy.cpp

```diff
@@ -1,18 +1,59 @@
-// RUN: %clangxx -%fsycl-host-only -c -ffp-accuracy=high -faltmathlib=SVMLAltMathLibrary -fno-math-errno %s
+// RUN: %clangxx -%fsycl-host-only -c -ffp-accuracy=high \
+// RUN: -faltmathlib=SVMLAltMathLibrary -fno-math-errno %s
+
+// RUN: %clangxx -fsycl %s -isystem %sycl_include/sycl -isystem %sycl_include \
+// RUN: -ffp-accuracy=high -fno-math-errno -S -emit-llvm -o- | FileCheck %s
+
+// RUN: %clangxx -fsycl %s -isystem %sycl_include/sycl -isystem %sycl_include \
+// RUN: -ffp-accuracy=high -ffp-accuracy=low:exp -fno-math-errno -S -emit-llvm \
+// RUN: -o- | FileCheck --check-prefix=CHECK-F1 %s
 
 #include <sycl/sycl.hpp>
 using namespace sycl;
 
+#define sz 2
+double input[sz];
+double output[sz];
+
 int main() {
   queue deviceQueue;
   double Value = 5.;
 
+  range<1> size{sz};
+
+  buffer<double, 1> in(input, size);
+  buffer<double, 1> out(output, size);
+
   deviceQueue.submit([&](handler &cgh) {
     cgh.single_task<class Kernel0>([=]() { double res = std::sin(Value); });
   });
 
   deviceQueue.submit([&](handler &cgh) {
     cgh.single_task<class Kernel1>([=]() { double res = sycl::sin(Value); });
   });
+
+  deviceQueue.submit([&](handler &cgh) {
+    accessor in_vals{in, cgh, read_only};
+    accessor out_vals{out, cgh, write_only};
+    cgh.single_task<class Kernel2>([=]() {
+      for (int i = 0; i < sz; i++)
+        out_vals[i] = cos(exp(log(in_vals[i])));
+    });
+  });
   return 0;
 }
+
+// CHECK-LABEL: define {{.*}}spir_kernel void @{{.*}}Kernel2
+// CHECK: tail call double @llvm.fpbuiltin.log.f64(double %2) #[[ATTR_HIGH:[0-9]+]]
+// CHECK: tail call double @llvm.fpbuiltin.exp.f64(double %3) #[[ATTR_HIGH]]
+// CHECK: tail call double @llvm.fpbuiltin.cos.f64(double %4) #[[ATTR_HIGH]]
```

Comment:
Are you trying to verify device or host LLVM IR? If the former, then it should go into https://github.com/intel/llvm/tree/sycl/sycl/test/check_device_code.

Reply:
Moved the test to test/check_device_code.